{
  RuntimeException runtimeException=new RuntimeException("Exception handle");
  when(selectedTrashFileItemsMock.toArray()).thenReturn(new TrashBrowseController.UIDeletedObject[]{uiDirectoryObjectMock});
  when(selectedTrashFileItemsMock.size()).thenReturn(1);
  doReturn(uiDirectoryObjectMock).when(selectedTrashFileItemsMock).get(0);
  doThrow(runtimeException).when(trashServiceMock).undelete(anyList());
  doReturn(false).when(mainControllerMock).handleLostRepository(any(Throwable.class));
  trBrController.undelete();
  verify(messageBoxMock).setTitle("Error");
  verify(messageBoxMock).setAcceptLabel("OK");
  verify(messageBoxMock).setMessage(contains("Exception handle"));
  verify(messageBoxMock,times(1)).open();
  verify(deckMock,never()).setSelectedIndex(1);
}
